src/utils/string/StringUtils.js

```diff
@@ -398,6 +397,10 @@ const onOpenLink = (url) => {
   });
 };
 
+const isValidUrl = (url) => {
+  return urlRegex.test(url);
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

While this function does check if the URL matches the regex, it doesn't reset the regex's lastIndex property. The `test()` method of a RegExp object with the global flag set retains the index at which it last matched within the string, and on subsequent calls starts searching from there. If you don't reset this, you might get incorrect results on successive calls.
```diff
- const isValidUrl = (url) => {
-   return urlRegex.test(url);
- };
+ const isValidUrl = (url) => {
+   const result = urlRegex.test(url);
+   urlRegex.lastIndex = 0;
+   return result;
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->